revalidation/initial_contacts/_01_icd_analysis.py

```diff
@@ -0,0 +1,305 @@
+    if int(get_env_var("MOBGAP_VALIDATION_USE_LOCAL_DATA", 0))
+    else None
+)
+loader = ValidationResultLoader(
+    "icd", result_path=local_data_path, version="main"
+)
+
+
+free_living_index_cols = [
+    "cohort",
+    "participant_id",
+    "time_measure",
+    "recording",
+    "recording_name",
+    "recording_name_pretty",
+]
+
+results = {
+    v: loader.load_single_results(k, "free_living")
+    for k, v in algorithms.items()
+}
+results = pd.concat(results, names=["algo", "version", *free_living_index_cols])
+results_long = results.reset_index().assign(
+    algo_with_version=lambda df: df["algo"] + " (" + df["version"] + ")",
+    _combined="combined",
+)
+cohort_order = ["HA", "CHF", "COPD", "MS", "PD", "PFF"]
+# %%
+# Performance metrics
+# -------------------
+# For each participant, performance metrics were calculated by classifying each sample in the recording as either
+# TP, FP, or FN.
+# Based on these values recall (sensitivity), precision (positive predictive value), F1 score were calculated.
+# On top of that the duration of overall detected initial contact per participant was calculated.
+# From this we calculate the mean and confidence interval for both systems, the bias and limits of agreement (LoA)
+# between the algorithm output and the reference data, the absolute error and the ICC.
+#
+# Below the functions that calculate these metrics are defined.
+from functools import partial
+
+from mobgap.pipeline.evaluation import CustomErrorAggregations as A
+from mobgap.utils.df_operations import (
+    CustomOperation,
+    apply_aggregations,
+    apply_transformations,
+)
+from mobgap.utils.tables import FormatTransformer as F
+
+custom_aggs = [
+    CustomOperation(
+        identifier=None,
+        function=A.n_datapoints,
+        column_name=[("n_datapoints", "all")],
+    ),
+    ("recall", ["mean", A.conf_intervals]),
+    ("precision", ["mean", A.conf_intervals]),
+    ("f1_score", ["mean", A.conf_intervals]),
+    ("ic_absolute_error_s", ["mean", A.loa]),
+    ("ic_relative_error", ["mean", A.loa]),
+]
+
+format_transforms = [
+    CustomOperation(
+        identifier=None,
+        function=lambda df_: df_[("n_datapoints", "all")].astype(int),
+        column_name=("General", "n_datapoints"),
+    ),
+    *(
+        CustomOperation(
+            identifier=None,
+            function=partial(
+                F.value_with_range,
+                value_col=("mean", c),
+                range_col=("conf_intervals", c),
+            ),
+            column_name=("ICD", c),
+        )
+        for c in [
+            "recall",
+            "precision",
+            "f1_score",
+        ]
+    ),
+    *(
+        CustomOperation(
+            identifier=None,
+            function=partial(
+                F.value_with_range,
+                value_col=("mean", c),
+                range_col=("loa", c),
+            ),
+            column_name=("IC Duration", c),
```

Comment:
Not happy with the term IC Duration. But also no better suggestion at the moment...